### PR TITLE
refactor: prefer using lo.BufferWithTimeout

### PR DESCRIPTION
**Pull request by Sidddddarth**
# Description

Using `lo.BufferWithTimeout` that either waits for a number of objects to receive from a channel or a timeout whichever's first.

## Notion Ticket

[BufferWithTimeout](https://github.com/samber/lo#bufferwithtimeout)

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: router/router.go [hunk 1]
```diff
@@ -1477,54 +1477,26 @@ func (rt *HandleT) commitStatusList(responseList *[]jobResponseT) {
...
-			lastUpdate = time.Now()
+			rt.commitStatusList(&jobResponseBuffer)
+			countStat.Count(numJobResponses)
 			statusStat.Since(start)
 		}
+		if !isResponseQOpen {
+			rt.logger.Debugf("[%v Router] :: statusInsertLoop exiting", rt.destName)
+			return
+		}
+		time.Sleep(maxStatusUpdateWait - timeToBuffer)
```
Comment (atzoum) [inline router/router.go:1499]: This doesn't take into consideration the time taken to commit. But overall, adding some extra sleep when the buffer is getting full faster than the `maxStatusUpdateWait` doesn't provide any actual benefit, on the contrary.

**Review by atzoum — COMMENTED** [review]
nice!

File: router/router.go (hunk 1 above)
Comment (Sidddddarth) [reply router/router.go:1499]: agreed
removing it

**Comment by atzoum** [comment]
@Sidddddarth is this still work in progress, or ready for review?

**Review by atzoum — APPROVED** [review]

**Review by cisse21 — APPROVED** [review]

**Pull request merged**